core/src/test/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -1504,8 +1503,8 @@ private void postIndexAsyncActions(String[] indices, List<CountDownLatch> inFlig
                     client().admin().indices().prepareFlush(indices).setIndicesOptions(IndicesOptions.lenientExpandOpen()).execute(
                             new LatchedActionListener<FlushResponse>(newLatch(inFlightAsyncOperations)));
                 } else if (isInternalCluster()) {
-                    internalCluster().getInstance(SyncedFlushService.class).attemptSyncedFlush(indices, IndicesOptions.lenientExpandOpen(),
-                            new LatchedActionListener<IndicesSyncedFlushResult>(newLatch(inFlightAsyncOperations)));
+                    client().admin().indices().syncedFlush(syncedFlushRequest(indices).indicesOptions(IndicesOptions.lenientExpandOpen()),
```

Comment:
You could probably drop the `isInternalCluster` check and instead give BWC tests some way to shut this off with more fine grained control. Because they can execute the synced flush if they are running against nodes at least as old as what you merge this into.